File: domusviewer/app/src/main/java/it/achdjian/paolo/domusviewer/DomusEngineRest/GetDevices.java
package it.achdjian.paolo.domusviewer.DomusEngineRest;

import android.content.SharedPreferences;
import android.os.Handler;
import android.support.annotation.NonNull;
import android.util.Log;

import com.fasterxml.jackson.core.type.TypeReference;
import com.fasterxml.jackson.databind.ObjectMapper;

import java.io.IOException;
import java.util.Map;

import it.achdjian.paolo.domusviewer.zigbee.ZDevices;

/**
 * Created by paolo on 15/04/16.
 */
public class GetDevices extends DomusEngineRest {
    private final ZDevices zDevices;

    public GetDevices(@NonNull SharedPreferences sharedPreferences, @NonNull ConnectionStatus connected, @NonNull ZDevices zDevices) {
        super(sharedPreferences, connected);
        this.zDevices = zDevices;
    }


    @Override
    public void run() {
        String body = get("/devices");
        if (body != null) {
            if (!body.isEmpty()) {
                ObjectMapper mapper = new ObjectMapper();
                try {
                    Map<String, String> devices = mapper.readValue(body, new TypeReference<Map<String, String>>() { });
                    for (Map.Entry<String, String> entry : devices.entrySet()) {
                        zDevices.addDevice(entry.getKey(), entry.getValue());
                    }
                } catch (IOException e) {
                    e.printStackTrace();
                }
            }
            new Handler().postDelayed(this, 2000);
        }
    }

}

File: temperature_monitor/app/src/main/java/it/achdjian/paolo/temperaturemonitor/domus_engine_rest/GetDevices.kt
package it.achdjian.paolo.temperaturemonitor.domus_engine_rest

import android.os.Handler
import com.fasterxml.jackson.core.type.TypeReference
import com.fasterxml.jackson.databind.ObjectMapper
import it.achdjian.paolo.temperaturemonitor.zigbee.ZDevices
import java.io.IOException

/**
 * Created by Paolo Achdjian on 7/9/17.
 */
class GetDevices(val domusEngineRest: DomusEngineRest, val zDevices: ZDevices) : Runnable {
    override fun run() {
        val body = domusEngineRest.get("/devices")
        if (body != null) {
            if (!body!!.isEmpty()) {
                val mapper = ObjectMapper()
                try {
                    val devices = mapper.readValue<Map<String, String>>(body, object : TypeReference<Map<String, String>>() {

                    })
                    for ((key, value) in devices) {
                        zDevices.addDevice(key, value)
                    }
                } catch (e: IOException) {
                    e.printStackTrace()
                }

            }
        }
    }
}